src/arrayMethodJoin.js

```diff
@@ -4,8 +4,24 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    for (let i = 0; i < this.length; i++) {
+      let copyElement = this[i];
+
+      if (this[i] == null) {
```

Comment:
Use strict equality '===' instead of '==' for comparison. And what will be in case this[i] will be undefined?

Reply:
when == used , undefined == null, but undefined !== null. I just didn't want to write the long condition if( this[i] === undefined || this[i] === null), should i use that anyway?